aiohttp/client.py

```diff
@@ -291,6 +315,9 @@ def _request(self, method, url, *,
                     # redirects
                     if resp.status in (
                             301, 302, 303, 307, 308) and allow_redirects:
+
+                        self.on_request_redirect.send(trace_context, resp)
```

Comment:
What is intended usage for the signal?
How to figure out what initial request was redirected?
The same is true for all other signals.
Maybe we should always pass URL, method and headers?

Reply:
The idea was to give a way to trace when a redirect happens. Completly agree that the signal parameters are not enough consistency to give enough information to the user, lets add the `method`, `URL`, `headers`.

What do you mean with :: ```The same is true for all other signals.```

Reply:
I will do the same with the `on_request_end`, and `on_request_exception` allowing the user to know the source URL in case the request came from a redirect.